scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -175,6 +176,196 @@ def _highly_variable_genes_seurat_v3(
         return df
 
 
+def _highly_variable_pearson_residuals(
+    adata: AnnData,
+    layer: Optional[str] = None,
+    n_top_genes: int = 2000,
+    batch_key: Optional[str] = None,
+    theta: float = 100,
+    clip: Union[Literal['auto', 'none'], float] = 'auto',
+    chunksize: int = 100,
+    subset: bool = False,
+    inplace: bool = True,
+) -> Optional[pd.DataFrame]:
+    """\
+    See `highly_variable_genes`.
+
+    Returns
+    -------
+    Depending on `inplace` returns calculated metrics (:class:`~pd.DataFrame`)
+    or updates `.var` with the following fields
+
+    highly_variable
+        boolean indicator of highly-variable genes
+    means
+        means per gene
+    variances
+        variances per gene 
+    residual_variances
+        Pearson residual variance per gene. Averaged in the case of multiple
+        batches.
+    highly_variable_rank
+        Rank of the gene according to residual variance, median rank in the
+        case of multiple batches
+    highly_variable_nbatches : int
+        If batch_key is given, this denotes in how many batches genes are
+        detected as HVG
+    highly_variable_intersection : bool
+        If batch_key is given, this denotes the genes that are highly variable
+        in all batches
+    """
+
+    view_to_actual(adata)
+    X = _get_obs_rep(adata, layer=layer)
```

Comment:
this line and view to actual, both methods are not defined I think, are you importing them somewhere?

Reply:
thanks for catching this, I imported them in `_normalize.py` and forgot to port to the hvg file as well. fixed.